gitpod-io/gitpod
# Image builds in prebuilds UI

## Description

Enables showing the logs of building the workspace image inside the prebuild detail page. 

<img width="948" alt="image" src="https://github.com/gitpod-io/gitpod/assets/29888641/cb89f742-dd22-42f2-abdc-56b511017b7d">


## Related Issue(s)

Fixes ENT-237

## How to test

1. Run a prebuild for an image, which the preview environment has not seen before
2. Observe the new tab dedicated to the image build logs. You should not be able to view any other tasks at that time and after its completion, it should disappear and show the first prebuild task.

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-imagebu86b0eea230</li>
	<li><b>🔗 URL</b> - <a href="https://ft-imagebu86b0eea230.preview.gitpod-dev.com/workspaces" target="_blank">ft-imagebu86b0eea230.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-imagebuilds-in-prebuild-ui-gha.26257</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-imagebu86b0eea230%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
